lib/matplotlib/artist.py

```diff
@@ -898,6 +900,89 @@ def set_zorder(self, level):
         self.pchanged()
         self.stale = True
 
+    def get_top_margin(self):
```

Comment:
I have a local branch (that I was working on last night before I ended up down that formatter rabbit hole #5594) that uses a single code path for `set`, `update`, and `setp`.

The `stale` logic is implemented via a property and should (maybe) also be whitelisted so we are at two (both of which are my fault).

We should probably provide both, use the property version internally and advertise the getter/setter version until 2.1.